### PR TITLE
[BEAM-2997] Encapsulate enums within a message so that C++/Python have meaningful namespaces when importing.

### PR DESCRIPTION
**Pull request by lukecwik**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
Test failure was unrelated in https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/14643/

retest this please

**Comment by lukecwik** [comment]
R: @robertwb or @kennknowles

**Comment by lukecwik** [comment]
TODO: Fix commit message before merging to state BEAM-2997 instead of BEAM-2959

**Comment by lukecwik** [comment]
retest this please

**Comment by lukecwik** [comment]
retest this please

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollectionTranslation.java [hunk 1]
```diff
@@ -62,19 +62,19 @@ public static IsBounded isBounded(RunnerApi.PCollection pCollection) {
...
-        return RunnerApi.IsBounded.UNBOUNDED;
+        return RunnerApi.IsBounded.Enum.UNBOUNDED;
       default:
         throw new IllegalArgumentException(
             String.format("Unknown %s %s", IsBounded.class.getSimpleName(), bounded));
     }
   }
 
-  static IsBounded fromProto(RunnerApi.IsBounded isBounded) {
+  static IsBounded fromProto(RunnerApi.IsBounded.Enum isBounded) {
```
Comment (kennknowles) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollectionTranslation.java]: I was really hoping for `IsBounded.IsBounded isBounded` and then maybe later we could even have assignments like `IsBounded.IsBounded isBounded = pcollection.isBounded()`

:-)

Comment (lukecwik) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollectionTranslation.java]: Me too, but then found this pattern in several proto users code and it made a lot of sense.

**Pull request merged**